### PR TITLE
allow kakfa custom serializer to be configured

### LINKED ISSUE #12960: kafka consumer: custom serializer can't be configured after it's instantiation
Please provide a detailed title (e.g. "Broker crashes when using TopN query with Bound filter" instead of just "Broker crashes").

### Affected Version

The Druid version where the problem was encountered.

### Description

KafkaRecordSupplier will take value.serializer to instantiate a custom serializer. 
however, some custom serializer(in my case, decryption serializer) may need to be configured by taking more config parameters from kafka parameters.

### PR DESCRIPTION
**Pull request by eshengit**
Fixes #12960.

#### Allow Kafka Consumer's custom deserializer to be configured after its instantiation. 

<hr>

##### Key changed/added classes in this PR
 * `extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java`

<hr>

This PR has:
- [ x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by eshengit** [comment]
I didn't fulfill all the PR requirement, I am waiting for the comments to see if the fix make sense for the majority use cases.

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java [hunk 1]
```diff
@@ -272,8 +280,8 @@ private static KafkaConsumer<byte[], byte[]> getKafkaConsumer(ObjectMapper sorti
     ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(KafkaRecordSupplier.class.getClassLoader());
-      Deserializer keyDeserializerObject = getKafkaDeserializer(props, "key.deserializer");
-      Deserializer valueDeserializerObject = getKafkaDeserializer(props, "value.deserializer");
+      Deserializer keyDeserializerObject = getKafkaDeserializer(props, "key.deserializer", true);
+      Deserializer valueDeserializerObject = getKafkaDeserializer(props, "value.deserializer", true);
```
Comment (abhishekagarwal87) [inline extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java:284]: shouldn't this be `false`?

**Review by abhishekagarwal87 — COMMENTED** [review]
your approach seems right to me. are you going to reopen this PR?

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java (hunk 1 above)
Comment (eshengit) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java:284]: ah, that's true!!! I will change it. thank you!

Comment (eshengit) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java:284]: @abhishekagarwal87 I just knew I need to go through our company's process to contribute to open source. I will reopen when I get approval. will keep you posted. thank you for reviewing.

Comment (abhishekagarwal87) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java:284]: Thank you.

**Comment by eshengit** [comment]
@abhishekagarwal87 , I created a new pr for the convenience. 
https://github.com/apache/druid/pull/13097 

the deleted repo gave some problems to sync the latest changes.

**Pull request closed without merging**